files/rsyslog/rsyslog.sh

```diff
@@ -42,109 +43,22 @@ issue_deprecation_warnings() {
-        if [ "${USE_MMEXTERNAL:-}" = "" ] ; then
-            # default - mmnormalize skip-empty
-            export SKIP_EMPTY=true
-            export USE_MMEXTERNAL=false
-            echo "INFO: SKIP_EMPTY was set to true and USE_MMEXTERNAL was not set, but the environment variables to manage undefined and empty fields were configured. Consider using USE_MMEXTERNAL=true and unsetting SKIP_EMPTY to activate the environment variables."
-        elif [ "${USE_MMEXTERNAL,,}" = "true" ] ; then
-            export SKIP_EMPTY=true
-            export USE_MMEXTERNAL=false
-            echo "INFO: Both USE_MMEXTERNAL and SKIP_EMPTY were set to true. Respecting SKIP_EMPTY. To activate the environment variables to manage undefined and empty fields configurations, consider unsetting SKIP_EMPTY."
-        else
-            export SKIP_EMPTY=true
-            export USE_MMEXTERNAL=false
-            echo "INFO: SKIP_EMPTY was set to true and USE_MMEXTERNAL was set to false, but the environment variables to manage undefined and empty fields were configured. Consider using USE_MMEXTERNAL=true and unsetting SKIP_EMPTY to activate the environment variables."
-        fi
-    else
-        if [ "${USE_MMEXTERNAL:-}" = "" ] ; then
-            export SKIP_EMPTY=false
-            export USE_MMEXTERNAL=true
-        elif [ "${USE_MMEXTERNAL,,}" = "true" ] ; then
-            export SKIP_EMPTY=false
-            export USE_MMEXTERNAL=true
-        else
-            export SKIP_EMPTY=false
-            export USE_MMEXTERNAL=true
-            echo "INFO: Both USE_MMEXTERNAL and SKIP_EMPTY were set to false, but the environment variables to manage undefined and empty fields were configured. Enabling USE_MMEXTERNAL. Consider using USE_MMEXTERNAL=true."
-        fi
-    fi
-fi
+# Undefined field configuration
+export USE_MMEXTERNAL=${USE_MMEXTERNAL:-true}
 
 undefined_conf="/var/lib/rsyslog.pod/undefined.json"
-if [ "${SKIP_EMPTY}" = "false" -a "${USE_MMEXTERNAL}" = "true" ] ; then
+if [ "${USE_MMEXTERNAL}" = "true" ] ; then
     # Create a config file for undefined fields from env vars
-    echo "{\"CDM_USE_UNDEFINED\" : ${CDM_USE_UNDEFINED}, \
+    echo "{\"CDM_USE_UNDEFINED\" : ${CDM_USE_UNDEFINED:-false}, \
 \"CDM_DEFAULT_KEEP_FIELDS\" : \"${CDM_DEFAULT_KEEP_FIELDS:-"CEE,time,@timestamp,aushape,ci_job,collectd,docker,fedora-ci,file,foreman,geoip,hostname,ipaddr4,ipaddr6,kubernetes,level,message,namespace_name,namespace_uuid,offset,openstack,ovirt,pid,pipeline_metadata,rsyslog,service,systemd,tags,testcase,tlog,viaq_msg_id"}\", \
 \"CDM_EXTRA_KEEP_FIELDS\" : \"${CDM_EXTRA_KEEP_FIELDS:-""}\", \
 \"CDM_UNDEFINED_NAME\" : \"${CDM_UNDEFINED_NAME:-"undefined"}\", \
-\"CDM_KEEP_EMPTY_FIELDS\" : \"${CDM_KEEP_EMPTY_FIELDS}\", \
-\"CDM_UNDEFINED_TO_STRING\" : ${CDM_UNDEFINED_TO_STRING}, \
-\"CDM_UNDEFINED_DOT_REPLACE_CHAR\" : \"${CDM_UNDEFINED_DOT_REPLACE_CHAR}\", \
+\"CDM_KEEP_EMPTY_FIELDS\" : \"${CDM_KEEP_EMPTY_FIELDS:-}\", \
```

Comment:
Do we want to set "message" to CDM_KEEP_EMPTY_FIELDS, by default? How about "kubernetes"?

Reply:
I don't think we want to add `kubernetes` by default - yet - let's see if people actually want all of the labels and annotations